tests/e2e/utils/kubeflow_installation.py

```diff
@@ -0,0 +1,200 @@
+    elif DEPLOYMENT_OPTION == "cognito":
+        path_dic = load_yaml_file(INSTALLATION_PATH_FILE_COGNITO)
+
+    print_banner(
+        f"You are installing kubeflow {DEPLOYMENT_OPTION} deployment with {INSTALLATION_OPTION}"
+    )
+
+    for component in Install_Sequence:
+        build_component(
+            INSTALLATION_OPTION,
+            DEPLOYMENT_OPTION,
+            component,
+            path_dic,
+            cluster_name,
+        )
+
+    if AWS_TELEMETRY_OPTION == "enable":
+        build_component(
+            INSTALLATION_OPTION,
+            DEPLOYMENT_OPTION,
+            "aws-telemetry",
+            path_dic,
+            cluster_name,
+        )
+
+
+def build_component(
+    INSTALLATION_OPTION,
+    DEPLOYMENT_OPTION,
+    component_name,
+    path_dic,
+    cluster_name,
+    crd_meet=True,
+    namespace=None,
+):
+    print(f"==========Installing {component_name}==========")
+    if component_name not in path_dic:
+        print(
+            f"component {component_name} is not applicable for deployment option: {DEPLOYMENT_OPTION}"
+        )
+        return
+    else:
+        installation_path = path_dic[component_name]["installation_options"][
+            INSTALLATION_OPTION
+        ]
+
+        if INSTALLATION_OPTION == "helm":
+            # cert-manager official chart command call
+            if component_name == "cert-manager":
+                build_retcode = build_certmanager()
+                assert build_retcode == 0
+            elif component_name == "aws-load-balancer-controller":
+                build_retcode = build_alb_controller(cluster_name)
+                assert build_retcode == 0
+            else:
+                install_helm(component_name, installation_path, namespace)
+        # kustomize
+        else:
+            if "crd_required" in path_dic[component_name]:
+                crd_required = path_dic[component_name]["crd_required"]
+                crd_meet = False
+            for kustomize_path in installation_path:
+                if not crd_meet:
+                    apply_kustomize(kustomize_path, crd_required)
+                    crd_meet = True
+                apply_kustomize(kustomize_path)
+
+    if "validations" in path_dic[component_name]:
+        identifier = path_dic[component_name]["validations"]["identifier"]
+        common_label = path_dic[component_name]["validations"]["common_label"]
+        namespace = path_dic[component_name]["validations"]["namespace"]
+        print(f"Waiting for {component_name} pods to be ready ...")
+        retcode = kubectl_wait_pods(common_label, namespace, identifier)
+        assert retcode == 0
+    print(f"All {component_name} pods are running!")
+
+
+def build_certmanager():
+    retcode = subprocess.call(
+        f"helm repo add jetstack https://charts.jetstack.io".split()
+    )
+    assert retcode == 0
+    retcode = subprocess.call(f"helm repo update".split())
```

Comment:
can we move the subprocess.call to a separate method so that assert ret_code == 0 doesnt have to be repeated so many times? feel free to write args to the functions like `run_cmd(cmd, assert=False/True)` to call it with and without asserting when needed

It will also be cleaner because the string split can happen in that method

Reply:
will do.